src/manager/agent.manager.service.ts

```diff
@@ -175,9 +191,22 @@ export class AgentManagerService {
      * for the existing one - we use the adminApiKey to ensure that the caller actually has permissions to query the agent
      * TODO we may want more checks here, eg to ensure the docker container is actually running, but for now we treat the cache as truth
      */
-    private async handleAlreadyRunningContainer(agentId: string, adminApiKey: string, autoConnect: boolean = true) {
+    private async handleAlreadyRunningContainer(agentId: string, adminApiKey: string, autoConnect: boolean = true, ttl: number): Promise<any> {
+        // TODO: cleanup inconsistent return types.
+        // 1 { agentId, connectionData }
+        // 2 { agentId, containerId, adminApiKey }
         if (agentId) {
-            const agentData: any = await this.cache.get(agentId);
+            let agentData: any = await this.cache.get(agentId);
+            if (agentData === undefined) {
+                Logger.warn(`agent ${agentId} not found in cache`);
+                // TODO  we dont really need containerId anymore, could remove it from cache
+                // @tothink() if we want to keep containerId, Dockerode contains a function to get containerId.  need
+                //   to research if k8s api has the same
+                const containerId = 'none';
+                await this.cache.set(agentId, { containerId, adminApiKey, ttl}, {ttl});
```

Comment:
I'm wondering if we only want to set this cache key if we successfully create the connection below - eg imagine it's the wrong adminApiKey being passed in we would cache it here even if the below call failed.

Reply:
took a slightly different approach for the same result.  I think this approach is a little cleaner